compiler/llvm/clangUtil.cpp

```diff
@@ -3531,6 +3531,52 @@ getCGArgInfo(const clang::CodeGen::CGFunctionInfo* CGI, int curCArg)
   return argInfo;
 }
 
+bool useDarwinArmFix(::Type* type) {
+  static bool isDarwin = strcmp(CHPL_TARGET_PLATFORM, "darwin") == 0;
+  static bool isArm = strcmp(CHPL_TARGET_ARCH, "arm64") == 0;
+
+  if (type != nullptr && isDarwin && isArm &&
+      type->symbol->hasFlag(FLAG_EXTERN) &&
+      isRecord(type)) {
+    return true;
+  }
+
+  return false;
+}
+
+//
+// Returns an ArgInfo for the given formal.
+//
+// This ArgInfo is acquired by manufacturing a function call in clang where the
+// formal is the only argument.
+//
```

Comment:
Right, sorry. The problem is that ``getClangABIInfo`` only works only works with extern types, so a function like this can't currently be processed:

```chpl
proc helper(in A : myExternStruct, B : ChapelRecord)
```

I believe this is ultimately due to the implementation of ``getClangType``, which will encounter a ``USR_FATAL`` when given something that isn't an extern type. I assumed that this as a limitation of our current interface with clang, and changing that seemed like a potentially large change (if that's even possible?).

If this is acceptable, I can add a comment that:
- emphasizes this is a workaround
- explains why ``getClangABIInfo`` isn't used

Reply:
(And a related point: it would be reasonable to apply something like this workaround for all functions that mix extern/export and non-extern/export argument types; so that might be a good TODO to note in the comment)